src/rulesets-new/json-api-rules/resource-object-rules.ts

```diff
@@ -0,0 +1,323 @@
+    responseBody.contentType === "application/vnd.api+json",
+  rule: (responseAssertions) => {
+    responseAssertions.body.added.matches({
+      schema: {
+        properties: {
+          jsonapi: {
+            type: Matchers.string,
+          },
+        },
+      },
+    });
+
+    responseAssertions.body.changed.matches({
+      schema: {
+        properties: {
+          jsonapi: {
+            type: Matchers.string,
+          },
+        },
+      },
+    });
+  },
+});
+
+const locationHeader = new ResponseRule({
+  name: "location header",
+  matches: (responseBody, rulesContext) =>
+    rulesContext.operation.method === "get" &&
+    responseBody.statusCode === "201",
+  rule: (responseAssertions) => {
+    responseAssertions.added.hasResponseHeaderMatching("location", {});
+    responseAssertions.changed.hasResponseHeaderMatching("location", {});
+  },
+});
+
+const selfLinks = new ResponseBodyRule({
+  name: "self links",
+  matches: (responseBody, rulesContext) => {
+    const { method } = rulesContext.operation;
+    const { statusCode } = responseBody;
+    if (method === "get" || method === "patch") {
+      return statusCode === "200";
+    } else if (method === "post") {
+      return statusCode === "201";
+    }
+    return false;
+  },
+  rule: (responseAssertions) => {
+    responseAssertions.body.added.matches({
+      schema: {
+        properties: {
+          links: {
+            properties: {
+              self: {},
+            },
+          },
+        },
+      },
+    });
+  },
+});
+
+const getPostResponseDataSchema = new ResponseBodyRule({
+  name: "valid get / post response data schema",
+  matches: (responseBody, rulesContext) => {
+    const { method } = rulesContext.operation;
+    const { statusCode, contentType } = responseBody;
+
+    return (
+      !isSingletonPath(rulesContext) &&
+      ["get", "post"].includes(method) &&
+      ["200", "201"].includes(statusCode) &&
+      contentType === "application/vnd.api+json"
+    );
+  },
+  rule: (responseAssertions) => {
+    const validSchemaShapes = [
+      {
+        schema: {
+          properties: {
+            data: {
+              type: "array",
+              items: {
+                properties: {
+                  id: {
+                    type: "string",
+                    format: "uuid",
+                  },
+                  type: {
+                    type: "string",
+                  },
+                },
+              },
+            },
+          },
+        },
+      },
+      {
+        schema: {
+          properties: {
+            data: {
+              properties: {
+                id: {
+                  type: "string",
+                  format: "uuid",
+                },
+                type: {
+                  type: "string",
+                },
+              },
+            },
+          },
+        },
+      },
+    ];
+
+    responseAssertions.body.added.matchesOneOf(validSchemaShapes);
```

Comment:
This effectively replaces this https://github.com/snyk/sweater-comb/blob/main/src/rulesets/jsonapi.ts#L373 and https://github.com/snyk/sweater-comb/blob/main/schemas/get-post-response-data.yaml